src/generic_ui/scripts/user.ts

```diff
@@ -41,46 +40,61 @@ module UI {
       }
       this.name = profile.name;
       this.imageData = profile.imageData || UI.DEFAULT_USER_IMG;
-      this.isOnline = profile.isOnline;
     }
 
     // Returns two strings, where each matches an array name in model.contacts.
     // Does not use an enum because we need a string value, and typescript
     // enums evaluate to numbers.
```

Comment:
w.r.t. console messages, I totally agree that using string representation is better and reviewers should check for this!